apps/vscode/command_client/vscode.py

```diff
@@ -0,0 +1,92 @@
+from pickle import FALSE
+from typing import Any
+
+from talon import Context, Module, actions
+
+from .command_client import NoFileServerException, NotSet, run_command
+
+ctx = Context()
+
+ctx.matches = r"""
+app: vscode
+"""
+ctx.tags = ["user.command_client"]
+mod = Module()
+
+
+def command_server_or_client_fallback(command_id: str, wait: bool):
+    """Execute command via command server and us command pallete if directory not present."""
+    try:
+        run_command(command_id, wait_for_finish=wait)
+    except NoFileServerException:
+        actions.user.command_palette()
+        actions.user.paste(command_id)
+        actions.key("enter")
+        print(
+            "Command issues via command palette for better performance install the VSCode extension for Talon"
+        )
+
+
+@ctx.action_class("user")
+class VsCodeAction:
+    def command_server_directory() -> str:
+        return "vscode-command-server"
+
+
+# These commands are shims, to provide backwards compatibility, they may be removed in the fuuture.
+# Prefer the run_command... version in command_client.
```

Comment:
This comment really applies to the last three actions. The first two will continue to be necessary so that we can still function without extension if they don't have it

Reply:
Moved to comment to correct position